src/pypowsybl_jupyter/networkexplorer.py

```diff
@@ -98,29 +105,29 @@ def go_to_vl_from_nad(event: any):
         vl_id= str(event.selected_node['equipment_id'])
         select_vl_and_activate_sld_tab(vl_id)
 
-    def update_nad_diagram(el):
-        nonlocal nad_widget
+    def compute_sld_data(el):
         if el is not None:
-            new_diagram_data=network.get_network_area_diagram(voltage_level_ids=el, 
-                                                              depth=selected_depth, high_nominal_voltage_bound=high_nominal_voltage_bound, 
-                                                              low_nominal_voltage_bound=low_nominal_voltage_bound, nad_parameters=npars)
-            if nad_widget==None:
-                nad_widget=display_nad(new_diagram_data, enable_callbacks=True)
-                nad_widget.on_select_node(lambda event : go_to_vl_from_nad(event))
-            else:
-                update_nad(nad_widget,new_diagram_data, enable_callbacks=True)
+            sld_data=network.get_single_line_diagram(el, spars)
+        else:
+            sld_data=EMPTY_SVG
+        return sld_data
+    
+    current_sld_data = compute_sld_data(None)
 
-    def update_sld_diagram(el, kv: bool = False):
+    def update_sld_widget(sld_diagram_data, kv: bool = False, enable_callbacks=True):
         nonlocal sld_widget
-        if el is not None:
-            sld_diagram_data=network.get_single_line_diagram(el, spars)
-            if sld_widget==None:
-                sld_widget=display_sld(sld_diagram_data, enable_callbacks=True)
-                sld_widget.on_nextvl(lambda event: go_to_vl(event))
-                sld_widget.on_switch(lambda event: toggle_switch(event))
+        if sld_widget==None:
+            sld_widget=display_sld(sld_diagram_data, enable_callbacks=enable_callbacks)
+            sld_widget.on_nextvl(lambda event: go_to_vl(event))
+            sld_widget.on_switch(lambda event: toggle_switch(event))
 
-            else:
-                update_sld(sld_widget, sld_diagram_data, keep_viewbox=kv, enable_callbacks=True)
+        else:
+            update_sld(sld_widget, sld_diagram_data, keep_viewbox=kv, enable_callbacks=enable_callbacks)
```

Comment:
Just realized that `keep_viewbox` parameter was not added in the sldwidget documentation, could you also add it?

Reply:
done